proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,670 @@
+During this process, no return message is spawned when a message is successfully delivered. 
+Error messages are only spawned by the Interoperability module when a message should be sent to an inactive chain, or if the receiving chain does not support the required logic.
+
+
+### Sending Cross-chain Messages
+
+Whenever a message is created, it must be added to the outbox of the corresponding partner chain. 
+On sidechains, this logic always appends to the mainchain outbox, while on the mainchain, this logic can append to any registered sidechain outbox.
+
+The Interoperability module exposes the <code>[send][base-interoperability-LIP-sendReducer]</code> reducer which should be used whenever other modules need to send cross-chain messages. 
+This functions also checks the liveness of the receiving chain, sets the message nonce property and appends the message to the outbox of the partner chain.
+
+
+### CCMs Specified by the Interoperability Module
+
+Then Interoperability module specifies the messages described in the following subsections.
+
+#### Cross-chain Update Receipt
+
+The main role of the cross-chain update receipt is to inform chains that a cross-chain update transaction has been posted on the partner chain, by whom, the fee that was paid, and the inbox size of the partner chain. 
+This is then used on the mainchain to allow for message recovery, see [LIP "Introduce sidechain recovery mechanism"][recovery-LIP]. 
+This can also be used in the sidechain to compensate the cross-chain update transaction poster, also called the _relayer_. 
+The precise choice of the compensation mechanism is left to sidechain developers. 
+For example, all cross-chain transaction fees could be split between the block forger and a compensation pool. 
+Whenever a cross-chain update transaction is posted on the mainchain, the poster gets compensation from the pool.
+
+
+#### Channel Terminated Message
+
+The role of the channel terminated message is to inform chains that their channel has been terminated on the mainchain. 
+The chain receiving this message can then also close the channel to the mainchain. 
+This is helpful in preventing users from sending transactions to a chain whilst the cross-chain update transaction will be invalid. 
+Note that the Interoperability module is designed in such a way that no channel should be terminated while the sidechain is respecting the Lisk interoperability protocol. 
+Sending and receiving this message should therefore be a rare occurrence.
+
+The termination message contains the last inbox size of the sidechain. 
+This allows the sidechain to know exactly which messages were executed on the mainchain and which are still pending at the time of termination.
+
+
+#### Registration Message
+
+The role of the registration message is to change the sending chain status from `CHAIN_REGISTERED` to `CHAIN_ACTIVE`. 
+Further, it guarantees that a channel was opened on the sidechain with the correct chain ID and network ID. 
```

Comment:
```suggestion
Further, it guarantees that a channel was opened on the sidechain with the correct name and network ID. 
```

Reply:
In reality you also explain this in the following sentences, so you could just drop this sentence

Reply:
removed the sentence

Reply:
Looking at the code, aren't all 3 values name, chain ID and network ID validated?